Pu bugfix

TEGELB opened the pull request:
Updates depreceated functions
might partly resolve #22

### Details of the Pull Request
- updated base_estimators to estimators due to impedding depreciation 
- updated the removed np.int 

Attempted to update removed `if_delegate_has_method ` with `available_if`, following code from the scikit-learn maintainers, still throws an attribute error I could not figure out. 
Elkanoto tests pass, Bagging still has this 1 error.

shaypal5 commented:
Hey @TEGELB ,

All bagging tests seem to fail with the following errors:
```python
self = <[AttributeError("'BaggingPuClassifier' object has no attribute 'base_estimator'") raised in repr()] BaggingPuClassifier object at 0x1b1c7ad17c8>
```

```python
E       TypeError: __init__() got an unexpected keyword argument 'estimator'
```

This seems to me to be directly related to your change of the following line:
```python
 super(BaseBaggingPU, self).__init__(
            base_estimator=base_estimator,
            n_estimators=n_estimators)
```
to
```python
 super(BaseBaggingPU, self).__init__(
            estimator=base_estimator,
            n_estimators=n_estimators)
```

Can you explain this specific change?

Because looking at the latest sklearn version, the `BaseForest` class for example, also extending the `BaseEnsemble` base class, much like our bagging classifiers, uses the `base_estimator` keyword argument, which from my understanding was not changed for ensemble classifiers:
```python
class BaseForest(MultiOutputMixin, BaseEnsemble, metaclass=ABCMeta):
    """
    Base class for forests of trees.

    Warning: This class should not be used directly. Use derived classes
    instead.
    """

    @abstractmethod
    def __init__(self,
                 base_estimator,
                 n_estimators=100, *,
                 estimator_params=tuple(),
                 bootstrap=False,
                 oob_score=False,
                 n_jobs=None,
                 random_state=None,
                 verbose=0,
                 warm_start=False,
                 class_weight=None,
                 max_samples=None):
        super().__init__(
            base_estimator=base_estimator,
            n_estimators=n_estimators,
            estimator_params=estimator_params)
```

TEGELB commented:
Hey @shaypal5,

thanks for the fast reply!

Yes, namely that the test suite passes through this future warning and I updated the occurences where future warnings were thrown to prevent issues when sklearn 1.4 is released:
![warning_temp](https://github.com/pulearn/pulearn/assets/56721736/eca8cd1b-306a-4e7c-9cc9-1df9fd8bdfbe)

Which sklearn version is used by the test, if it predates the change this would explain an error being thrown?
The Base Estimator Class itself has the future warning in it:

https://github.com/scikit-learn/scikit-learn/blob/7f9bad99d6e0a3e8ddf92a7e5561245224dab102/sklearn/ensemble/_base.py#L78

```python
class BaseEnsemble(MetaEstimatorMixin, BaseEstimator, metaclass=ABCMeta):
    """Base class for all ensemble classes.

    Warning: This class should not be used directly. Use derived classes
    instead.

    Parameters
    ----------
    estimator : object
        The base estimator from which the ensemble is built.

    n_estimators : int, default=10
        The number of estimators in the ensemble.

    estimator_params : list of str, default=tuple()
        The list of attributes to use as parameters when instantiating a
        new base estimator. If none are given, default parameters are used.

    base_estimator : object, default="deprecated"
        Use `estimator` instead.

        .. deprecated:: 1.2
            `base_estimator` is deprecated and will be removed in 1.4.
            Use `estimator` instead.

    Attributes
    ----------
    estimator_ : estimator
        The base estimator from which the ensemble is grown.

    base_estimator_ : estimator
        The base estimator from which the ensemble is grown.

        .. deprecated:: 1.2
            `base_estimator_` is deprecated and will be removed in 1.4.
            Use `estimator_` instead.

    estimators_ : list of estimators
        The collection of fitted base estimators.
    """
```  

I've reverted to base_classifier so the tests can run through as is. 


I furthermore found the error in the bagging suite and fixed it.

shaypal5 commented:
Also note you have two `flake8` errors.

Please lint using `flake8` and fix any linting errors before updating the PR. 🙏🏽

shaypal5 commented:
Great. So only Python 3.7 tests fail, and it's past EOL, so I'm gonna drop these anyway once we merge.

Only thing left is to get test coverage back to 100%.

You should be getting the coverage report when running locally, so it should be easy to fix.

Almost there! 🥇 💪🏽

TEGELB commented:
Hello @shaypal5 ,

found the issues, now at 100% coverage and local tests pass. 

Cheers

shaypal5 commented:
Released in `v0.0.8` 🎉 

Amazing job, @TEGELB !
Thank you for your contribution!

shaypal5 commented:
You are now credited in the README :)
